src/snowflake/snowpark/dataframe_writer.py

```diff
@@ -265,10 +266,26 @@ def copy_into_location(
         header: bool = False,
         statement_params: Optional[Dict[str, str]] = None,
         block: Literal[False] = False,
-        **copy_options: Optional[Dict[str, Any]],
+        **copy_options: Unpack[CopyOptions],
     ) -> AsyncJob:
         ...  # pragma: no cover
 
+    @overload
+    def copy_into_location(
+        self,
+        location: str,
+        *,
+        partition_by: Optional[ColumnOrSqlExpr] = None,
+        file_format_name: Optional[str] = None,
+        file_format_type: Optional[str] = None,
+        format_type_options: Optional[Dict[str, str]] = None,
+        header: bool = False,
+        statement_params: Optional[Dict[str, str]] = None,
+        block: bool = True,
+        **copy_options: Unpack[CopyOptions],
+    ) -> Union[List[Row], AsyncJob]:
+        ...  # pragma: no cover
+
```

Comment:
The type hint of a function definition without overload did not work with csv method or json method, and a pylancen error occurred, so I also defined it with overload.